f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py

```diff
@@ -233,9 +305,118 @@ def _pre_create(self, service):
             self.driver.network_builder.prep_service_networking(
                 service, traffic_group)
 
+    def __get_update_operation(self, old_loadbalancer, loadbalancer):
+
+        # the logic is
+        # if both are not zero are are not equal then modify
+        # if old isn't zeor while new is zero, delete.
+        # if new isn't zeor while old is zero, add.
+        if 'bandwidth' not in old_loadbalancer.keys():
+            if 'bandwidth' in loadbalancer.keys() and loadbalancer['bandwidth']:
+                return 'Add'
+        else:
+            old_value = old_loadbalancer['bandwidth']
+            if old_value:
+                if 'bandwidth' in loadbalancer.keys() and \
+                    loadbalancer['bandwidth']:
+                    if loadbalancer['bandwidth'] != old_loadbalancer['bandwidth']:
+                        return 'Modify'
+                else:
+                    return 'Delete'
+            else:
+                if 'bandwidth' in loadbalancer.keys() and \
+                    loadbalancer['bandwidth']:
+                    return 'Add'
+
+        return 'None'
+
+    def __update_listener_bwc(self, bigip, loadbalancer, service, add=True):
+
+        if 'listeners' in service.keys():
+            irule_name = self.__get_irule_profile_name(loadbalancer)
+            for listener in service['listeners']:
+                vs_payload = self.driver.service_adapter.\
+                    _init_virtual_name(loadbalancer, listener)
+                # add is True then add irule to vs and otherwise delete irule
+                # from vs
+                vs = self.vs_helper.load(bigip, name=vs_payload['name'],
+                                    partition=vs_payload['partition'])
+                vs_payload['rules'] = vs.rules
+                if add is True:
+                    if irule_name not in vs_payload['rules']:
+                        vs_payload['rules'].append(irule_name)
+                else:
+                    if irule_name in vs_payload['rules']:
+                        vs_payload['rules'].remove(irule_name)
+                self.vs_helper.update(bigip, vs_payload)
+        return
+
+    def __add_bwc(self, bigip, loadbalancer, service, update_listener=True):
+        # The logic is , add the irule bwc policies and go through listeners to
+        # bind the bwc policy.
+        bandwidth = loadbalancer.get('bandwidth')
+        bandwidth = bandwidth * 1000 * 1000
```

Comment:
1024 or 1000 ?

Reply:
qin, I tested that if we use 1024, the value configured in bigip is not correct. So I changed to 1000 and it worked well.